File: 268. Missing Number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        allnums = [i for i in xrange(len(nums)+1)]
        res = allnums[0]
        for i in xrange(1,len(allnums)):
            res ^= allnums[i]
        for i in xrange(len(nums)):
            res ^= nums[i]
        return res

File: 268. Missing Number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        vector <int> allnums;
        for (int i = 0; i<nums.size()+1; i++)
            allnums.push_back(i);
        int res = allnums[0];
        for (int i = 1; i<allnums.size(); i++)
            res ^= allnums[i];
        for (int i = 0; i<nums.size(); i++)
            res ^= nums[i];
        return res;
    }
};
